CMakePresets.json

```diff
@@ -0,0 +1,183 @@
+{
+    "version": 3,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 16,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "common",
+            "hidden": true,
+            "description": "Common farm-ng configuration",
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/out/build/${presetName}",
+            "installDir": "${sourceDir}/out/install/${presetName}",
+            "cacheVariables": {
+                "BUILD_SOPHUS_TESTS": "Off",
+                "CMAKE_TOOLCHAIN_FILE": "${sourceDir}/vcpkg/scripts/buildsystems/vcpkg.cmake",
+                "CMAKE_CXX_FLAGS": "-Werror -Wall -Wextra -Wpedantic -Wno-unused-local-typedef -Wno-unused-local-typedef -Wno-gnu-zero-variadic-macro-arguments -Wno-unused-parameter -Wno-unused-variable -Wno-vla-extension",
+                "CMAKE_CXX_EXTENSIONS": "OFF",
+                "CMAKE_CXX_STANDARD": "20",
+                "CMAKE_CXX_STANDARD_REQUIRED": "ON",
+                "CMAKE_CXX_FLAGS_RELWITHDEBINFO": "-O2 -g",
+                "FARM_NG_PROVIDER_DEV_PACKAGES": "CLI11;Eigen3;Sophus"
```

Comment:
Why express this here instead of in the `CMakeLists.txt` file where it's invoked?

Reply:
The `CMakeLists.txt` job is just to express what dependencies the package depends on. It's the choice of an individual developer / packager where they want to get those dependencies from. For example, a developer might have a preset in their personal `CMakeUserPresets.json` that lets them switch between using the development and released versions of Sophus.